File: src/main/java/miragefairy2019/mod3/manualrepair/api/IManualRepairableItem.java
package miragefairy2019.mod3.manualrepair.api;

import net.minecraft.item.ItemStack;
import net.minecraft.item.crafting.Ingredient;
import net.minecraft.util.NonNullList;

public interface IManualRepairableItem {

    public boolean canManualRepair(ItemStack itemStack);

    public NonNullList<Ingredient> getManualRepairSubstitute(ItemStack itemStack);

    public ItemStack getManualRepairedItem(ItemStack itemStack);

}

File: src/main/java/miragefairy2019/mod3/manualrepair/api/IManualRepairableItem.kt
package miragefairy2019.mod3.manualrepair.api

import net.minecraft.item.ItemStack
import net.minecraft.item.crafting.Ingredient

interface IManualRepairableItem {
    fun canManualRepair(itemStack: ItemStack): Boolean
    fun getManualRepairIngredients(itemStack: ItemStack): List<Ingredient>
    fun getManualRepairedItem(itemStack: ItemStack): ItemStack
}
